pmd-java/src/main/resources/category/java/bestpractices.xml

```diff
@@ -209,6 +209,36 @@ class Foo {
         </example>
     </rule>
 
+    <rule name="AvoidReassigningCatchVariables"
+          since="6.27.0"
+          message="Avoid reassigning caught exception ''{0}''"
+          class="net.sourceforge.pmd.lang.java.rule.bestpractices.AvoidReassigningCatchVariablesRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_bestpractices.html#avoidreassigningcatchvariables">
+        <description>
+Reassigning exception variables caught in a catch statement should be avoided because of:
+1) If it is needed, multi catch can be easily added and code will still compile
+2) Following the principle of least surprise we want to make sure that a variable caught in a catch statement is always the one thrown in a try block
+        </description>
+        <priority>2</priority>
```

Comment:
I'll change the priority to 3 when merging.

We - by now - don't suggest any priorities and 3 is in the middle. It is up to the user to decide, how important a rule for the project is...